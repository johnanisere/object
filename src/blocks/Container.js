import styled from "styled-components";

const Container = styled.section`
  display: flex;
  padding: 20px 10px;
  @media (max-width: 768px) {
    flex-direction: column;
  }
`;
export default Container;
