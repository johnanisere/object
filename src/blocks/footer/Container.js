import styled from "styled-components";

const Container = styled.div`
  -webkit-tap-highlight-color: transparent;
  box-sizing: border-box;
  text-align: ${({ align }) => align || "left"};
  flex: ${({ flex }) => (flex ? flex : 1)};
`;

export default Container;
