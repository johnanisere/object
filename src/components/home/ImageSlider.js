import React from "react"
import PropTypes from "prop-types"
import Image from "../../blocks/Image"
import Container from "../../blocks/FlexibleContainer";


const App = ({ data, onSwitchImage, active }) => {
    return (
        <Container background={true} slider={true}>
            <button>{`<`}</button>
            <div>
                {
                    data.map(({ displayImage }, key) => <Image onClick={() => onSwitchImage(key)} active={key !== active} key={key}><img alt="me" src={displayImage} /></Image>)
                }
            </div>
            <button>></button>
        </Container>
    )
}

App.propTypes = {
    data: PropTypes.array.isRequired,
    active: PropTypes.number.isRequired,
    onSwitchImage: PropTypes.func.isRequired
}
export default App