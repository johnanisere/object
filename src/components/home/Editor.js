import React from "react"
import PropTypes from "prop-types"
import ImageSlider from "./ImageSlider"
import Container from "../../blocks/FlexibleContainer";


class App extends React.Component {
    render() {
        const { handleMouseMove, handleClick, displayData, cursor, element, data, onSwitchImage, active, displayImage } = this.props;
        return (
            <Container background={true} height="100%" flex={3} border={true} direction="column">
                <Container header={true} />
                <Container editor={true} cursor={cursor} >

                    <figure className="img-holder" onClick={handleClick} onMouseMove={handleMouseMove} >
                        <img src={displayImage} alt="The Pulpit Rock" />

                        {
                            displayData.map(({ Figure }, key) => <Figure key={key} ></Figure>)
                        }
                        {element && element}
                    </figure>
                </Container>
                <ImageSlider data={data} onSwitchImage={onSwitchImage} active={active} />
            </Container>
        );
    }
}


App.propTypes = {
    onSwitchImage: PropTypes.func.isRequired,
    handleMouseMove: PropTypes.func.isRequired,
    displayImage: PropTypes.string.isRequired,
    handleClick: PropTypes.func.isRequired,
    displayData: PropTypes.array.isRequired,
    cursor: PropTypes.string.isRequired,
    active: PropTypes.number.isRequired,
    data: PropTypes.array.isRequired,
    element: PropTypes.node,
}
export default App