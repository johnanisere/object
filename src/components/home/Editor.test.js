
import React from "react";
import { shallow } from "enzyme";

import Editor from "./Editor"
import ImageSlider from "./ImageSlider"
import Container from "../../blocks/FlexibleContainer";

const props = {
    handleMouseMove: () => { },
    onSwitchImage: () => { },
    displayImage: "",
    handleClick: () => { },
    displayData: [],
    cursor: "",
    data: [],
    active: 1,
}

it("Editor component exists", () => {
    const wrapper = shallow(<Editor {...props} />);
    expect(wrapper.exists()).toBe(true);
});

it("Editor component renders", () => {
    const wrapper = shallow(<Editor {...props} />);
    expect(wrapper).toMatchSnapshot();
});

it(`Editor should contain "Container",  "ImageSlider", `, () => {
    const wrapper = shallow(<Editor {...props} />);

    expect(wrapper.find('figure')).toHaveLength(1);
    expect(wrapper.find('img')).toHaveLength(1);
    expect(wrapper.find(Container)).toHaveLength(3);
    expect(wrapper.find(ImageSlider)).toHaveLength(1);
});