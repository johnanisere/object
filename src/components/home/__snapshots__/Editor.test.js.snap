// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor component renders 1`] = `ShallowWrapper {}`;
